source/Irrlicht/CNullDriver.cpp

```diff
@@ -192,6 +193,54 @@ void main()
         addBuiltInToCaches(sampler, "irr/builtin/samplers/default_clamp_to_border");
 	}
 
+    //images
+    core::smart_refctd_ptr<asset::ICPUImage> dummy1x1Image;
+    {
+        asset::ICPUImage::SCreationParams info;
+        info.format = asset::EF_R8_UNORM;
+        info.type = asset::ICPUImage::ET_2D;
+        info.extent.width = 1u;
+        info.extent.height = 1u;
+        info.extent.depth = 1u;
+        info.mipLevels = 1u;
+        info.arrayLayers = 1u;
+        info.samples = asset::ICPUImage::ESCF_1_BIT;
+        info.flags = static_cast<asset::IImage::E_CREATE_FLAGS>(0u);
+        dummy1x1Image = core::make_smart_refctd_ptr<asset::ICPUImage>(std::move(info));
+
+        auto buf = core::make_smart_refctd_ptr<asset::ICPUBuffer>(1u);
+        reinterpret_cast<int8_t*>(buf->getPointer())[0] = 0;
+        auto regions = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<asset::ICPUImage::SBufferCopy>>(1u);
+        asset::ICPUImage::SBufferCopy& region = regions->front();
+        region.imageSubresource.mipLevel = 0u;
+        region.imageSubresource.baseArrayLayer = 0u;
+        region.imageSubresource.layerCount = 1u;
+        region.bufferOffset = 0u;
+        region.bufferRowLength = 1u;
+        region.bufferImageHeight = 0u;
+        region.imageOffset = {0u, 0u, 0u};
+        region.imageExtent = {1u, 1u, 1u};
+        dummy1x1Image->setBufferAndRegions(std::move(buf), regions);
+    }
+    
+    //image views
+    {
```

Comment:
great but wrong file for all this stuff, IAssetManager or similar

no idea why youre screwing around with this in the `video` namespace

Reply:
idk, thats where all default assets are added and you started this :P

Reply:
I don't think I started this, I'm not a great fan of lambdas.

Only thing I remember is a default up and down streaming buffer.

OFC there was the 4 default materials/shaders that I've carried on from vanilla-irrlicht, but they started that (and you should nuke them, now! - replace with a default pipeline that does a textured, solid non opaque thing with no lighting).

Everything needs to move to IAssetManager or some more appropriate place within `irr::asset`

maybe to reduce clutter we should make a `IAssetManager::createBuiltinsBeforeLoaders` and `IAssetManager::createBuiltinsAfterLoaders` to call before and after the asset-loaders and registered and constructed.

Reply:
that said `IVideoDriver` (or CPU to GPU asset converter) could benefit from a method called `convertAllBuiltinsToGPU` for memory reclaiming

Reply:
did you move this?

Reply:
yea